src/payment/strategies/ppsdk/step-handler/continue-handler/redirect/RedirectionState.ts

```diff
@@ -0,0 +1,43 @@
+export const PENDING_REDIRECT_PARAM = 'redirecting';
+
+export class RedirectionState {
+    private _isRedirecting: boolean;
+
+    constructor() {
+        this._isRedirecting = this.getCurrentUrl().searchParams.has(PENDING_REDIRECT_PARAM);
+    }
+
+    private getCurrentUrl(): URL {
+        return new URL(window.location.href);
```

Comment:
I believe we still support IE11 in storefront, so we can't use the native `URL` provided by browsers. You can however import `parseUrl` from `common/url` and use that instead if you prefer to deal with objects.

Reply:
I've moved to using `parseUrl` from `common/url`: https://github.com/bigcommerce/checkout-sdk-js/pull/1285/commits/90b45fd57c67b7b5edc14081193e13606906917e

Though `parseUrl` doesn't have any real support for parsing search params, which has led to me doing a bit of regex and string manipulation within `RedirectionState` - more than I'd ideally like.

Depending on wider opinion, I'd be open to removing this ad hoc manipulation and moving to either:

- Using polyfill for `URL` and replacing all uses of `parseUrl` from `common/url`
- Swapping `common/url` for a more fully featured, open source package like https://github.com/unshiftio/url-parse

Reply:
I'm happy for you to go with either approach (use an OSS so it's more fully-featured). If we go with polyfill, it'd be good to pick one that doesn't modify the global `window`.

Reply:
Polyfilling the `URL` was a non starter: `core-js` provides only a global altering version, `core-js-pure` provides an isolated version, doesn't provide types. All other deviate too far from the official spec to be worth while.

Instead we've gone for `query-string`, which just seeks to help manage url queries. It's not as full featured as `URL` but does move us to manipulating objects and not manipulating strings with regex and ad hoc concatenations: https://github.com/bigcommerce/checkout-sdk-js/pull/1285/commits/361e9211c3570a6d989da22f12d41f11763f7767

Which is then used in `RedirectionState`: https://github.com/bigcommerce/checkout-sdk-js/pull/1285/commits/60f7403fb21fe4efc91cdf88be1192d2bce90355